packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -214,6 +215,30 @@ class Config:
     """User inputs expected by this provider when initializing it. Each `Field` `f`
     should be passed in the constructor as a keyword argument, keyed by `f.key`."""
 
+    manages_history: ClassVar[bool] = False
+    """Whether this provider manages its own conversation history upstream. If
+    set to `True`, Jupyter AI will not pass the chat history to this provider
+    when invoked."""
+
+    persona: ClassVar[Optional[Persona]] = None
+    """
+    The **persona** of this provider, a struct that defines the name and avatar
+    shown on agent replies in the chat UI. When set to `None`, `jupyter-ai` will
+    choose a default persona when rendering agent messages by this provider.
+
+    Because this field is set to `None` by default, `jupyter-ai` will render a
+    default persona for all providers that are included natively with the
+    `jupyter-ai` package. This field is reserved for Jupyter AI modules that
```

Comment:
Why is it set to `None` rather than `JupyternautPersona` then?

Reply:
We may want to make the default persona configurable in the future. However, the provider classes themselves are not traitlets-configurable. By allowing `ConfigManager` to choose the default instead of hard-coding the default in `BaseProvider`, we allow for the default persona to potentially be configurable in the future.